sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -264,6 +264,10 @@ object NullPropagation extends Rule[LogicalPlan] {
       case e @ Substring(_, Literal(null, _), _) => Literal.create(null, e.dataType)
       case e @ Substring(_, _, Literal(null, _)) => Literal.create(null, e.dataType)
 
+      // MaxOf and MinOf can't do null propagation
```

Comment:
`MaxOf(null, null) == null` but `MaxOf(null, value) != null`, that's why we can't do null propagation here. And `MaxOf(null, null)` will be optimized in `ConstantFolding`, we don't need to handle it here.